### PR TITLE
Chris/integrate p2p stream

### PR DESCRIPTION
**Pull request by CHr15F0x**
Fixes: #1329

### CODE REVIEW

File: crates/p2p/src/client/peer_agnostic.rs [hunk 1]
```diff
@@ -182,32 +189,18 @@ impl Client {
...
-                }
-                // Try the next peer instead
-                Err(error) => {
-                    tracing::debug!(from=%peer, %error, "bodies request failed");
+            if let Ok(rx) = response_receiver {
+                if let Some(parsed) = parse::<parse::state_update::State>(&peer, rx).await {
+                    return Ok(parsed);
                 }
             }
+            // Try the next peer
```
Comment (pierre-l) [inline crates/p2p/src/client/peer_agnostic.rs:203]: Leftover?

File: crates/p2p/src/client/peer_agnostic.rs [hunk 2]
```diff
@@ -239,35 +232,20 @@ impl Client {
...
-                        tracing::debug!(from=%peer, "empty response or unexpected end of response");
-                        break;
-                    }
-                }
-                // Try the next peer
-                Err(error) => {
-                    tracing::debug!(from=%peer, %error, "transactions request failed");
+            if let Ok(rx) = response_receiver {
+                if let Some(parsed) = parse::<parse::transactions::State>(&peer, rx).await {
+                    return Ok(parsed);
```
Comment (pierre-l) [inline crates/p2p/src/client/peer_agnostic.rs:245]: A match seems simpler to me than `map_err` + `if let Ok`

File: crates/p2p/src/main_loop.rs [hunk 3]
```diff
@@ -399,152 +412,182 @@ impl MainLoop {
...
-                        .send(Event::InboundTransactionsSyncRequest {
-                            from: peer,
-                            request,
-                            channel,
-                        })
-                        .await
-                        .expect("Event receiver not to be dropped");
-                }
-                request_response::Message::Response {
+                let _ = self
```
Comment (pierre-l) [inline crates/p2p/src/main_loop.rs:512]: Do we really not care about the result here? If so, this could use a comment ;)

**Review by Mirko-von-Leipzig — APPROVED** [review]

File: crates/p2p/src/main_loop.rs [hunk 4]
```diff
@@ -399,152 +412,182 @@ impl MainLoop {
...
-                        .send(Event::InboundReceiptsSyncRequest {
-                            from: peer,
-                            request,
-                            channel,
-                        })
-                        .await
-                        .expect("Event receiver not to be dropped");
-                }
-                request_response::Message::Response {
+                let _ = self
```
Comment (pierre-l) [inline crates/p2p/src/main_loop.rs:547]: Same here, expect or comment why ignored

**Review by pierre-l — APPROVED** [review]

File: crates/p2p/src/client/peer_agnostic.rs (hunk 1 above)
Comment (CHr15F0x) [reply crates/p2p/src/client/peer_agnostic.rs:203]: I meant:
```
// Else: try the next peer
```

File: crates/p2p/src/main_loop.rs (hunk 3 above)
Comment (CHr15F0x) [reply crates/p2p/src/main_loop.rs:512]: There's plenty of places where the same comment would have to be added ("if the sending fails the client is dropped and we cannot do much about it").

**Pull request merged**